edgelet/edgelet-kube/src/module/create.rs

```diff
@@ -29,18 +30,37 @@ where
     S::Error: Fail,
     S::Future: Send,
 {
+    let runtime_for_owner = runtime.clone();
+
     let runtime_for_sa = runtime.clone();
     let module_for_sa = module.clone();
 
+    let runtime_for_rb = runtime.clone();
+    let module_for_rb = module.clone();
+
     let runtime_for_deployment = runtime.clone();
     let module_for_deployment = module.clone();
 
     let module_name = module.name().to_string();
+    let iotedged_module_name = "iotedged";
 
-    create_or_update_service_account(&runtime, &module)
-        .and_then(move |_| create_or_update_role_binding(&runtime_for_sa, &module_for_sa))
-        .and_then(move |_| {
-            create_or_update_deployment(&runtime_for_deployment, &module_for_deployment)
+    get_module_owner(&runtime_for_owner, &iotedged_module_name)
+        .and_then(move |module_owner| {
+            let module_owner1 = module_owner.clone();
```

Comment:
I think another approach here, you can wrap module_owner into Arc and place cloning references when call methods.